src/mon/commands/authmon_cmds.cc

```diff
@@ -0,0 +1,162 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2019 SUSE LLC <contact@suse.com> 
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software 
+ * Foundation.  See file COPYING.
+ * 
+ */
+
+#include "mon/Monitor.h"
+#include "mon/AuthMonitor.h"
+#include "mon/MonOpRequest.h"
+
+#include "mon/commands/Command.h"
+#include "mon/commands/authmon_cmds.h"
+
+#include "messages/MMonCommand.h"
+
+#include "include/types.h"
+#include "include/ceph_assert.h"
+#include "include/Context.h"
+
+#include "common/dout.h"
+#include "common/cmdparse.h"
+
+#define dout_subsys ceph_subsys_mon
+
+
+// read commands
+//
+
+bool AuthMonAuthExport::do_preprocess(
+    MonOpRequestRef op,
+    const string &prefix,
+    const cmdmap_t &cmdmap,
+    stringstream &ss,
+    bufferlist rdata,
+    FormatterRef f)
+{
+  string entity_name;
+  EntityName entity;
+  bool ret = AuthMonCommon::get_entities(cmdmap, entity, entity_name, ss);
+  if (!ret) {
+    reply(op, -EINVAL, ss, get_last_committed());
+    return true;
```

Comment:
It would be nice to make the return type something other than bool (see much longer discussion above on the general interface). This /looks/ like "hey this was invalid, so we clearly mean to fail... and return true!" even though it is really just saying "eh, it's fine". And below, it happens again (line 68) and also at the end of the function. Having something like "result::ok" or "result::not_failure" etc. is worth thinking about. It also fits into the desire to use std::error_code that has come up before. 

Again, this might be something that you may want to handle incrementally as it does involve high-touch on the existing implementation of commands, but it's worth thinking about.